src/bm_sim/calculations.cpp

```diff
@@ -360,36 +360,110 @@ struct csum16 {
   }
 };
 
+// Concatenate the next n bytes into the specified type.
+// Pad zeros at tail when given bytes are shorter than the type.
+// For example, concat<uint32_t>(&{0xab, 0xcd}, 2) == 0xabcd0000.
+template <typename T>
+T concat(const char *buf, size_t n) {
```

Comment:
I don't think `concat` is a great name for this function. It's not doing any concatenation. Maybe `emitAsInteger`?

Reply:
Sure, will rename it.